File: sdks/js/src/exceptions.ts
export class VeklomError extends Error {
  constructor(message: string) {
    super(message);
    this.name = 'VeklomError';
  }
}

export class AuthError extends VeklomError {
  constructor(message: string) {
    super(message);
    this.name = 'AuthError';
  }
}

export class RateLimitError extends VeklomError {
  constructor(message: string) {
    super(message);
    this.name = 'RateLimitError';
  }
}

export class BudgetExceededError extends VeklomError {
  constructor(message: string) {
    super(message);
    this.name = 'BudgetExceededError';
  }
}

File: sdks/python/veklom/exceptions.py
"""Veklom SDK exceptions."""


class VeklomError(Exception):
    """Base exception for all Veklom SDK errors."""


class AuthError(VeklomError):
    """Raised on 401 / 403 responses."""


class RateLimitError(VeklomError):
    """Raised on 429 responses."""


class BudgetExceededError(VeklomError):
    """Raised when workspace budget cap is hit."""


class ModelUnavailableError(VeklomError):
    """Raised when the requested model is offline and no fallback is available."""
